packages/sakuli-legacy/src/context/sahi/action/mouse-actions-api.function.spec.ts

```diff
@@ -3,150 +3,163 @@ import {By, ThenableWebDriver} from "selenium-webdriver";
-            const api = createApi(driver);
-            await driver.get(mockHtml(`
+        describe('mouse interaction', () => {
+            type MouseMethods = "_click" | "_rightClick" | "_mouseOver" | "_mouseUp" | "_mouseDown";
+            it.each(<[MouseMethods, string, string][]>[
+                ['_click', 'click', ""],
+                ['_rightClick', 'contextmenu', ""],
+                ['_mouseOver', 'mouseover', ""],
+                ['_mouseUp', 'mouseup', ""],
+                ['_mouseDown', 'mousedown', ""],
+                // ['_click', 'click', "CTRL"],
+                ['_click', 'click', "META|ALT"],
+                ['_rightClick', 'contextmenu', "META|ALT"],
+                ['_mouseOver', 'mouseover', "META"],
+                ['_mouseUp', 'mouseup', "SHIFT|ALT"],
+                ['_mouseDown', 'mousedown', "ALT"],
+            ])('%s should invoke native event %s with %s pressed', async (method: MouseMethods, nativeEvent: string, combo: string) => {
+                const api = createApi(driver);
+                await driver.get(mockHtml(`
+            <style>
+              #btn:hover {background: red;} 
+            </style>
             <button id="btn">Click Me</button>
             <div id="out"></div>
             <script >
               const $$ = document.getElementById.bind(document);
               const btn = $$('btn');
               const out = $$('out');
               btn.addEventListener('${nativeEvent}', e => {
+                  e.preventDefault();
                   const keyAddon = [];
                   if(e.ctrlKey) keyAddon.push('CTRL');
                   if(e.metaKey) keyAddon.push('META');
                   if(e.shiftKey) keyAddon.push('SHIFT');
                   if(e.altKey) keyAddon.push('ALT');
                   out.innerHTML = '${method} ' + keyAddon.join("|");
+                  return false;
               });
             </script>
         `));
-            const apiMethod: any = api[method];
-            if (apiMethod.length === 2) {
-                await apiMethod({
-                        locator: By.css('#btn'),
-                        identifier: 0,
-                        relations: []
-                    },
-                    combo);
-            } else {
-                await apiMethod({
-                        locator: By.css('#btn'),
-                        identifier: 0,
-                        relations: []
-                    },
-                    false,
-                    combo);
-            }
-
-            const out = await driver.findElement(By.css('#out'));
-            return expect(out.getText()).resolves.toBe(`${method} ${combo}`.trim());
+                const apiMethod: any = api[method];
+                const query: SahiElementQuery = {
+                    locator: By.css('#btn'),
+                    identifier: 0,
+                    relations: []
+                };
+                if(method === '_mouseUp') {
```

Comment:
What's the reasoning for mapping `_mouseUp` to `_mouseDown`?

Reply:
It's actually not a mapping, it rather invokes a mouseDown if the method to test is mouseUp. Reason for this is that firefox doesnt emit a mouseup event without a previous mousedown.

TBH: There should be a dedicted test for this rather the applying logic to this testcase

Reply:
Agreed, I think it would be better to keep browser specific stuff in separate tests in a way that allows us to detect if some of this specific stuff changes in the future.

Reply:
According to the `_mouseUp` implementation, this looks like a false positive to me.
`_mouseUp` only calls `a => a.move(toElement(e)).release(mouseButton)`, so there wouldn't be a preceding `mousedown` event in a Sakuli test with FireFox.

Reply:
the problem is that only running ` a => a.move(toElement(e)).release(mouseButton)` will not fire the `mouseup` event in ff so the testcase checks if it is testing `_mouseUp` and invokes a mouseDown before. In this case FF fires the mousedown event in dom

Reply:
i'll update the tests

Reply:
The thing is that calling `_mouseUp` in a Sakuli test would result in a NOOP on FireFox, even though CI says it works as expected, since we manually did a `_mouseDown` beforehand.

Reply:
Conclusion: Will be explicitly pointed out in docs when working with FireFox. Test will be left as is, since it proofs that the event will be fired when handled correctly.